quarto/_extensions/countdown/countdown.lua

```diff
@@ -0,0 +1,295 @@
+-- Specify embedded version 
+local countdownEmbeddedVersion = "0.0.1"
+
+-- Only embed resources once if there are multiple timers present 
+local needsToExportDependencies = true
```

Comment:
I believe Quarto deduplicate dependencies (see first paragraph in https://quarto.org/docs/extensions/lua-api.html#html-dependencies ). So you can declare the same dependency more than once and only one copy will be included. So you could simplify this a little bit and just add the countdown does every time.

Reply:
We need the custom catch because we're injecting into the header the global style as part of a text include. To reduce the check from re-happening with the add_html_dependency() and to group the actions, I placed it alongside the header injection.

Reply:
Oh I see, that makes sense